doc/visual_testing/reporter_visual_inspection_suite.py

```diff
@@ -376,298 +367,166 @@ def report_surface_flm(build_type):
 
 
-def report_surface_slm(build_type):
-    if build_type == "partial":
-        _generate_dummy_html(filenames=["slm_surf_empty.html"])
-        return None
-
+def report_surface_slm():
     slm = SecondLevelModel(mask_img=SurfaceMasker())
     report_slm_empty = slm.generate_report(height_control="bonferroni")
     report_slm_empty.save_as_html(REPORTS_DIR / "slm_surf_empty.html")
 
     return report_slm_empty
 
 
-# %%
-# Adapted from examples/03_connectivity/plot_probabilistic_atlas_extraction.py
-def report_nifti_maps_masker(build_type):
-    if build_type == "partial":
-        _generate_dummy_html(filenames=["nifti_maps_masker.html"])
-        return None
+# %%%%%%%%%% MASKER REPORTS %%%%%%%%%%
 
-    atlas = fetch_atlas_difumo(dimension=64, resolution_mm=2)
-    atlas_filename = atlas["maps"]
 
-    data = fetch_development_fmri(n_subjects=1)
+def _generate_masker_report_files_partial(
+    masker, **kwargs
+) -> tuple[HTMLReport, None]:
+    """Generate masker report files for partial doc build.
 
-    masker = NiftiMapsMasker(
-        maps_img=atlas_filename,
-        standardize="zscore_sample",
-        standardize_confounds=True,
-        memory="nilearn_cache",
-        cmap="gray",
-        memory_level=1,
-        reports=False,
-    )
-    report = masker.generate_report(
-        title="Reporting disabled and Unfitted Nifti Maps Masker Report"
+    Should generate reports for:
+    - unfitted masker
+    - unfitted masker with reports=False
+
+    Returns
+    -------
+    tuple[HTMLReport, None]
+        First element is the unfitted masker report.
+    """
+    masker_class_name = masker.__class__.__name__
+
+    unfitted_report = masker.generate_report(
+        title=f"{masker_class_name} unfitted", **kwargs
     )
-    report.save_as_html(
-        REPORTS_DIR / "nifti_maps_masker_no_reporting_unfitted.html"
+    unfitted_report.save_as_html(
+        REPORTS_DIR / f"{masker_class_name}_unfitted.html"
     )
 
-    masker.reports = True
-    report = masker.generate_report(title="Unfitted Nifti Maps Masker Report")
-    report.save_as_html(REPORTS_DIR / "nifti_maps_masker_unfitted.html")
-
     masker.reports = False
-    masker.fit(data.func[0])
-    report = masker.generate_report(
-        title="Reporting disabled - Fitted Nifti Maps Masker Report"
+    unfitted_report_no_reporting = masker.generate_report(
+        title=f"{masker_class_name} unfitted - reports=False", **kwargs
     )
-    report.save_as_html(
-        REPORTS_DIR / "nifti_maps_masker_no_reporting_fitted.html"
+    unfitted_report_no_reporting.save_as_html(
+        REPORTS_DIR / f"{masker_class_name}_unfitted_reports-False.html"
     )
 
-    masker.reports = True
-    masker.fit(data.func[0])
-    print(masker._reporting_data)
-    report = masker.generate_report(
-        title="Nifti Maps Masker - Fitted with Development FMRI",
-        displayed_maps=[2, 6, 7],
-    )
-    report.save_as_html(REPORTS_DIR / "nifti_maps_masker.html")
```

Comment:
@hndgzkn I have sort of removed this check with report=False from this script (i.e fitting with reports=False, setting it to True and then asking for a report)

- I think it should have a test for it instead in https://github.com/nilearn/nilearn/pull/5836

Reply:
I've added the test to https://github.com/nilearn/nilearn/pull/5836.